platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -541,6 +551,17 @@ def get_xcvr_api(self):
                 self._xcvr_api.get_tx_fault = self.get_tx_fault
         return self._xcvr_api
 
+    def is_sw_control(self):
+        if not DeviceDataManager.is_independent_mode():
+            return False
+
+        db = utils.DbUtils.get_db_instance('STATE_DB')
```

Comment:
@Junchao-Mellanox why do we need to update this in STATE_DB? Static platform specific file is not sufficient?

Reply:
This PR will read state DB to understand which cable is under software control/firmware control. Static platform file is not accurate because:

On init stage, static platform file always indicate software control, but the module could be in "Not initialize" state.

Reply:
@Junchao-Mellanox From what I understand, CMIS module is under SW control and non-CMIS are under FW control. There is a TRANSCEIVER_INFO table which can be used to know CMIS vs non-CMIS. Why not use that?

Reply:
Will use TRANSCEIVER_INFO table.